src/components/NextPageMiddleware.tsx

```diff
@@ -82,7 +82,17 @@ const NextPageMiddleware = ({ children }: Props) => {
         }
 
         if (isProfileReady && profile.wallets().count() === 0) {
-            navigate('/splash-screen');
+            const lastVisitedPage = profile.settings().get(ProfileData.LastVisitedPage) as
+                | LastVisitedPage
+                | undefined;
+
+            lastVisitedPage || persistScreen ? navigate('/onboarding') : navigate('/splash-screen');
+
+            if (lastVisitedPage) {
+                navigate('/wallet/create');
+                return;
+            }
```

Comment:
yes, it is correct, we are navigating to `onboarding` then to the last screen because if don't the back button will act up, that is why we need to push a route to the history. Yeah, it is not pleasant, but it works.

Reply:
worth a comment in the file if it's for a specific scenario like that

Reply:
I prefer a comment even if it's temporary, as that avoids scenarios where the temporary fix stays around for longer than planned